image_data_guide/03a_builtin_preprocessing.ipynb

```diff
@@ -0,0 +1,600 @@
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Creating .rec file from /home/ec2-user/SageMaker/sagemaker-notebooks-WIP/sagemaker_image_data_guide/data_recordio/train.lst in /home/ec2-user/SageMaker/sagemaker-notebooks-WIP/sagemaker_image_data_guide/data_recordio\n",
+      "time: 0.2049543857574463  count: 0\n",
+      "time: 5.9312732219696045  count: 1000\n",
+      "time: 5.552445888519287  count: 2000\n"
+     ]
+    }
+   ],
+   "source": [
+    "!python im2rec.py --resize 224 --quality 90 --num-thread 16 data_recordio/train data_structured/train"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Validation dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Creating .rec file from /home/ec2-user/SageMaker/sagemaker-notebooks-WIP/sagemaker_image_data_guide/data_recordio/val.lst in /home/ec2-user/SageMaker/sagemaker-notebooks-WIP/sagemaker_image_data_guide/data_recordio\n",
+      "time: 0.0018703937530517578  count: 0\n"
+     ]
+    }
+   ],
+   "source": [
+    "!python im2rec.py --resize 224 --quality 90 --num-thread 16 data_recordio/val data_structured/val"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<pre>\n",
+    "</pre>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Upload the data to S3"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Create a bucket for your project"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Bucket Name: sagemaker-prebuilt-ic-2a789e07-853b-4b89-b0eb-8ea33550520e\n"
+     ]
+    }
+   ],
+   "source": [
+    "if pathlib.Path('pickled_data/prebuilt_bucket_name.pickle').exists():\n",
+    "    with open('pickled_data/prebuilt_bucket_name.pickle', 'rb') as f:\n",
+    "        bucket_name = pickle.load(f)\n",
+    "        print('Bucket Name:', bucket_name)\n",
+    "else:\n",
+    "    bucket_name = f'sagemaker-prebuilt-ic-{str(uuid.uuid4())}'\n",
+    "    s3 = boto3.resource('s3')\n",
+    "    region = sagemaker.Session().boto_region_name\n",
+    "    bucket_config = {'LocationConstraint': region}\n",
+    "    s3.create_bucket(Bucket=bucket_name, CreateBucketConfiguration=bucket_config)\n",
+    "    \n",
+    "    with open('pickled_data/prebuilt_bucket_name.pickle', 'wb') as f:\n",
+    "        pickle.dump(bucket_name, f)\n",
+    "    print('Bucket Name:', bucket_name)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Upload .rec files to S3"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**NEEDS UPDATING**"
```

Comment:
What needs updating?

Reply:
I must have forgotten to remove that note. Note has been removed.